### PR TITLE
reward model training

### PR DESCRIPTION
**Pull request by apbard**

### CODE REVIEW

File: examples/nanoChatGPT/model.py
Comment (tcbegley) [inline examples/nanoChatGPT/model.py:1]: Is this different to the one inside `nanoGPT`? If not we can just import like [here](https://github.com/tcbegley/rl/blob/nanoChatGPT/examples/nanoChatGPT/train.py#L30) right?

Comment (apbard) [reply examples/nanoChatGPT/model.py:1]: nope, but I think we would need to change them also (e.g. move the init/setup etc within the model) or clean_up a bit

Comment (tcbegley) [reply examples/nanoChatGPT/model.py:1]: We could have our own model.py where we import and make some modifications, but making changes inside the nanoGPT repo itself starts to get more complicated. 

I think it's also beneficial for ther user to be able to compartmentalise: "everything in nanoGPT is unchanged, so I just need to understand the other stuff"

File: examples/nanoChatGPT/train_reward.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+        batch = next(train_loader)
+
+        # TODO: check why is different from std model (missing micro gradients)
+        
+        # TODO: combine evaluate_loss function with this. it's almost the same thing
+        # evaluate the loss
+        reward_chosen = model(batch.chosen)
+        reward_rejected = model(batch.rejected)
+        loss = -torch.log(torch.sigmoid(reward_chosen.reward - reward_rejected.reward)).mean()
```
Comment (tcbegley) [inline examples/nanoChatGPT/train_reward.py:385]: Since we wrapped model with `TensorDictModule` above, we should be passing in the tensorclass here, rather than a tensor.

But also the model expects the attribute `input` so we need to do something like

```python
batch_chosen = batch.select("chosen", "reward").rename_key_("chosen", "input")
batch_rejected = batch.select("rejected", "reward").rename_key_("rejected", "input")

model(batch_chosen)
model(batch_rejected)
```

Then we can rewrite `loss` as 

```python
loss = -torch.log(torch.sigmoid(batch_chosen.reward - batch_rejected.reward)).mean()
```

Comment (tcbegley) [reply examples/nanoChatGPT/train_reward.py:385]: This does feel a little bit clunky though

Comment (apbard) [reply examples/nanoChatGPT/train_reward.py]: the code above is not working as reward_chosen and reward_rejected are a tensor.
Instead of creating the tensordict we can just remove the ".reward" from the loss.
